README.md

```diff
@@ -144,81 +144,146 @@ The old repo is no longer the recommended approach for running and extending the
 
-2. From one terminal, run the agent:
+Once you are done modifying the `.1env` file, run the following: 
 
-    ```bash
-    bash run_agent.sh
-    ```
+```bash
+# Source the env file
+source .1env
+```
 
-3. From another terminal, run the Tendermint node:
+Finally, in the file `run_service.sh`, change the line `autonomy deploy build -ltm` to `autonomy deploy build -ltm -n k`,
+where k is replaced by the number of agents in your service. For instance for 1 agent: `autonomy deploy build -ltm -n 1`.
 
-    ```bash
-    bash run_tm.sh
-    ```
+The rest of the common environment variables are present in the [service.yaml](https://github.com/valory-xyz/mech/blob/main/packages/valory/services/mech/service.yaml), which are customizable too.
 
-### Option 2: Run the Mech as an agent service
+#### Running the service 
 
-1. Ensure you have a file with the agent address and private key (`keys.json`). You can generate a new private key file using the Open Autonomy CLI:
 
-    ```bash
-    autonomy generate-key ethereum -n 1
+1. Run the service:
+
     ```
+    poetry shell
+    poetry install
+    bash run_service.sh
+    ```
+
+2. Build the docker container by running the following:
+
+```
+cd mech
+build_dir=$(ls -d abci_build_????/ 2>/dev/null || echo "abci_build")
+autonomy deploy run --build-dir $build_dir
+```
+
+
+3. In a separate terminal, run the following to see the logs:
+
+```
+container = $(basename "$(ls -d ../*/abci_build_????/ 2>/dev/null | head -n 1)" | sed -E 's/^abci_build_//')
+container = "mech${container}_abci_0"
+docker logs -f container
+```
+
+### Option 2: Run the Mech as a standalone agent
+
+#### Configuration of the agent
 
-2. Ensure that the variable `ALL_PARTICIPANTS` in the file `.1env` contains the agent address from `keys.json`:
+First, ensure you have a file with a private key (`ethereum_private_key.txt`). You can generate a new private key file using the Open Autonomy CLI:
 
-   ```bash
-   ALL_PARTICIPANTS='["your_agent_address"]'
+   ```
+   autonomy generate-key ethereum 
    ```
 
-3. Run, the service:
+Replace the key by the one of your agent's address (you need to have an EOA for this).
 
-    ```bash
-    bash run_service.sh
+Then you need to configure the agent. You need to create a `.agentenv` file which contains the service configuration parameters. We provide a prefilled template (`.example_agent.env`). You will need to use an [OpenAI API key](https://platform.openai.com/account/api-keys) in the configuration.
+
+Run the following, where `network` is replaced by the name of the network (gnosis or base):
+
+```bash
+# Copy the prefilled template
+cp .example_agent_network.env .agentenv
+```
+
+The variables to change are the same as for the [service](#configuration-of-the-service). Other customizable variables can be found in the `packages/valory/agents/mech/aea-config.yaml` file. 
```

Comment:
If we were to keep this part, we should also explain how to extract the env var's name.